src/shogun/mathematics/linalg/ratapprox/tracesampler/ProbingSampler.cpp

```diff
@@ -197,7 +197,7 @@ SGVector<float64_t> CProbingSampler::sample(index_t idx) const
 	{
 		if (m_coloring_vector[i]==idx)
 		{
-			float64_t x=sg_rand->std_normal_distrib();
+			float64_t x=std_normal_distrib();
```

Comment:
I wonder whether `m_rand->std_random()` wouldnt be cleaner? I.e. make clear that this is a method from the random.....? Then the mixin would add that member instead of all the methods ... but not sure whether this is better than what you did here.....

Reply:
yup it's actually better imo. don't know why i didn't do that :")

Reply:
nope ... i disagree with it. rather follow PRNG concept requirements for prng in standard, namely have a class for each distribution and overload `operator()` that way we'll be able to use those in STL containers and algos...

Reply:
@vigsterkr explaned this to me. the random mixin will have the sole job of hosting a prng object, and providing a functionality for setting seed, so CRandom class is not needed.